src/js/components/FileInput/doc.js

```diff
@@ -0,0 +1,142 @@
+import { describe, PropTypes } from 'react-desc';
+
+import { getAvailableAtBadge } from '../../utils/mixins';
+import { themeDocUtils } from '../../utils/themeDocUtils';
+
+export const doc = FileInput => {
+  const DocumentedFileInput = describe(FileInput)
+    .availableAt(getAvailableAtBadge('FileInput'))
+    .description('A control to input one or more files.')
+    .usage(
+      `import { FileInput } from 'grommet';
+<FileInput id='file' name='file' />`,
+    )
+    .intrinsicElement('input');
+
+  DocumentedFileInput.propTypes = {
+    accept: PropTypes.string.description(
+      'MIME type pattern to match against. For example: "image/*".',
+    ),
+    aggregateThreshold: PropTypes.number
+      .description(
+        `The maximum number of individual files to show. Above this,
+      only a single message describing the number of files will be shown.`,
```

Comment:
I think it should include that this prop is relevant only when using `multiple`.
And it might be worth to simplify it by removing this prop and make it embedded in the `multiple` prop functionality, so multiple may accept a number or object.

Reply:
I think passing a number to `multiple` would initially be interpreted as either the required number of files or the maximum number of files, neither of which are right.

Reply:
I've updated the description to reference that it only applies when multiple is true.

Reply:
One other approach would be to support something like `multiple={{ aggregateAt: 10 }}`. Thoughts?

Reply:
YES for `multiple={{ aggregateAt: 10 }}` I like it better than how it is now.
Overall, I don't think the current aggregation property should be used much and it's better to keep the API clean with one less prop if possible.

Reply:
Changed as requested :)